pages/admin/reports/components/TotalEarnings.tsx

```diff
@@ -0,0 +1,245 @@
+import { Box, Text, useBreakpointValue } from '@chakra-ui/react';
+import {
+    BarChart,
+    ResponsiveContainer,
+    Bar,
+    XAxis,
+    YAxis,
+    Tooltip,
+    Legend,
+} from 'recharts';
+import { useTreatments } from '@/hooks/useTreatments';
+import { useEffect, useState } from 'react';
+import { getAppointments } from '@/services/AppointmentService';
+import { Appointment } from '@/interfaces/appointment/Appointment';
+import { splitDateTimeFromISO } from '@/utils/dateUtils';
+import {
+    TotalEarningsProps,
+    TreatmentStats,
+} from '@/interfaces/statistics/TotalEarnings';
+import parseTreatmentPrice from '@/utils/priceUtils';
+
+export default function TotalEarnings({
+    statistics,
+    dateRange,
+    selectedTreatmentId,
+}: TotalEarningsProps) {
+    const { treatments } = useTreatments();
```

Comment:
💭   I see that you're fetching the treatments here and in the DateInputs component, this means you're doing the same request twice to display a view. Is it possible to use this hook in the index (Reports) component and send the treatments by props?.

This way you can reuse the treatments in both components and do the error handling in the same Index component rather than in DateInputs